server/apps/import_text_archive/app_initialize.py

```diff
@@ -0,0 +1,250 @@
+                item['type'] = 'text'
+
+            item['keyword'] = self._get_head_value(doc, 'Keyword')
+            item['ingest_provider_sequence'] = self._get_head_value(doc, 'Sequence')
+
+            orginal_source = self._get_head_value(doc, 'Author')
+            if orginal_source:
+                item['original_source'] = orginal_source
+
+            item['headline'] = self._get_head_value(doc, 'Headline')
+
+            code = self._get_head_value(doc, 'SubjectRefNum')
+            if code and len(code) == 7:
+                code = '0' + code
+            if code and code in subject_codes:
+                item['subject'] = []
+                item['subject'].append({'qcode': code, 'name': subject_codes[code]})
+                try:
+                    process_iptc_codes(item, None)
+                except:
+                    pass
+
+            slug = self._get_head_value(doc, 'SLUG')
+            if slug:
+                item['slugline'] = slug
+            else:
+                item['slugline'] = self._get_head_value(doc, 'Keyword')
+
+            # self._addkeywords('Takekey', doc, item)
+            take_key = self._get_head_value(doc, 'Takekey')
+            if take_key:
+                item['anpa_take_key'] = take_key
+
+            self._addkeywords('Topic', doc, item)
+
+            self._addkeywords('Selectors', doc, item)
+
+            el = doc.find('dcdossier/document/body/BodyText')
+            if el is not None:
+                story = el.text
+                if item['type'] == 'text':
+                    story = story.replace('\n   ', '<br><br>')
+                    story = story.replace('\n', '<br>')
+                    item['body_html'] = story
+                else:
+                    item['body_html'] = story
+                try:
+                    item['word_count'] = get_text_word_count(item['body_html'])
+                except:
+                    pass
+
+            item['pubstatus'] = 'usable'
+
+            res = superdesk.get_resource_service('text_archive')
+            original = res.find_one(req=None, guid=item['guid'])
+            if not original:
+                res.post([item])
+            else:
+                res.patch(original['_id'], item)
+
+            if self._limit:
+                self._limit -= 1
+            # print(item)
+
+    def run(self, start_id, user, password, url, query, limit):
+        print('Starting text archive import at {}'.format(start_id))
+        self._user = user
+        self._password = password
+        self._id = int(start_id)
+        self._url_root = url
+        self._query = urllib.parse.quote(query)
+        self._limit = int(limit)
+
+        self._api_login()
+
+        x = self._get_bunch(self._id)
+        while x:
+            self._process_bunch(x)
+            x = self._get_bunch(self._id)
+            if self._limit is not None and self._limit <= 0:
+                break
+
+        print('finished text archive import')
```

Comment:
No need, this populates both mongo and elastic, might be a problem in it's self! So far the performance is limited by DC/REST not elastic. I guess this may change as the dataset gets big.